[2024-07-17T01:19:52.210+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-17T01:19:52.228+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T05:10:00+00:00 [queued]>
[2024-07-17T01:19:52.233+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T05:10:00+00:00 [queued]>
[2024-07-17T01:19:52.233+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-17T01:19:52.240+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): etl_usd_mxn> on 2024-07-15 05:10:00+00:00
[2024-07-17T01:19:52.245+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=30238) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-07-17T01:19:52.245+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_ticker', 'etl_usd_mxn', 'scheduled__2024-07-15T05:10:00+00:00', '--job-id', '613', '--raw', '--subdir', 'DAGS_FOLDER/bid_ask_dag.py', '--cfg-path', '/tmp/tmpkffg78oz']
[2024-07-17T01:19:52.249+0000] {standard_task_runner.py:63} INFO - Started process 30241 to run task
[2024-07-17T01:19:52.249+0000] {standard_task_runner.py:91} INFO - Job 613: Subtask etl_usd_mxn
[2024-07-17T01:19:52.280+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_ticker.etl_usd_mxn scheduled__2024-07-15T05:10:00+00:00 [running]> on host 6f712de23803
[2024-07-17T01:19:52.340+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_ticker' AIRFLOW_CTX_TASK_ID='etl_usd_mxn' AIRFLOW_CTX_EXECUTION_DATE='2024-07-15T05:10:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-15T05:10:00+00:00'
[2024-07-17T01:19:52.340+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-17T01:19:52.349+0000] {logging_mixin.py:188} WARNING -   0%|          | 0/600 [00:00<?, ?it/s]
[2024-07-17T01:19:53.510+0000] {bid_ask_dag.py:87} INFO - Threshold! {'orderbook_timestamp': '2024-07-17T01:19:51+00:00', 'book': 'usd_mxn', 'bid': 17.61, 'ask': 17.64, 'spread': 0.1700680272108908}
[2024-07-17T01:19:53.516+0000] {logging_mixin.py:188} WARNING -   0%|          | 0/600 [00:01<?, ?it/s]
[2024-07-17T01:19:53.518+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-17T01:19:53.519+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/bid_ask_dag.py", line 88, in etl_spread
    SlackWebhookHook(webhook_token=WEBHOOK_URL, message=f"Alert! bid-ask has appeared {record}").execute()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: SlackWebhookHook.__init__() missing 1 required keyword-only argument: 'slack_webhook_conn_id'
[2024-07-17T01:19:53.537+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_ticker, task_id=etl_usd_mxn, run_id=scheduled__2024-07-15T05:10:00+00:00, execution_date=20240715T051000, start_date=20240717T011952, end_date=20240717T011953
[2024-07-17T01:19:53.553+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 613 for task etl_usd_mxn (SlackWebhookHook.__init__() missing 1 required keyword-only argument: 'slack_webhook_conn_id'; 30241)
[2024-07-17T01:19:53.598+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-17T01:19:53.609+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-17T01:19:53.610+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
